src/storage/mod.rs

```diff
@@ -1134,14 +1134,28 @@ impl<E: Engine> Storage<E> {
         limit: usize,
         statistics: &mut Statistics,
         key_only: bool,
+        reverse: bool,
     ) -> Result<Vec<Result<KvPair>>> {
         let mut option = IterOption::default();
         if let Some(end) = end_key {
-            option.set_upper_bound(end.into_encoded());
+            if reverse {
```

Comment:
I am not very enjoy `if reverse` every where here, which is not good in both looking and performance. Do you have some better ideas? @MyonKeminta How about split into two functions?

Reply:
Good idea

Reply:
should I define raw_scan_reverse?

Reply:
Maybe `reverse_raw_scan`, I think

Reply:
split raw_scan function